app/controllers/vatapplication/ApplyForEoriController.scala

```diff
@@ -40,22 +45,37 @@ class ApplyForEoriController @Inject()(val sessionService: SessionService,
   val show: Action[AnyContent] = isAuthenticatedWithProfile {
     implicit request =>
       implicit profile =>
-        vatApplicationService.getVatApplication map {
-          _.eoriRequested match {
-            case Some(eoriRequested) => Ok(applyForEoriView(ApplyForEoriForm.form.fill(eoriRequested)))
-            case None => Ok(applyForEoriView(ApplyForEoriForm.form))
-          }
+        vatRegistrationService.partyType flatMap {
+          partyType =>
+            vatApplicationService.getVatApplication map {
+              _.eoriRequested match {
+                case Some(eoriRequested) => Ok(getView(partyType, ApplyForEoriForm.form.fill(eoriRequested)))
+                case None => Ok(getView(partyType, ApplyForEoriForm.form))
+              }
+            }
         }
+
   }
 
   val submit: Action[AnyContent] = isAuthenticatedWithProfile {
     implicit request =>
       implicit profile =>
-        ApplyForEoriForm.form.bindFromRequest.fold(
-          errors => Future.successful(BadRequest(applyForEoriView(errors))),
-          success => vatApplicationService.saveVatApplication(EoriRequested(success)) map { _ =>
-            Redirect(controllers.vatapplication.routes.TurnoverEstimateController.show)
-          }
-        )
+        vatRegistrationService.partyType flatMap {
```

Comment:
Could this be moved into the "errors" part of the fold, doesn't look like the success uses it so we don't want an unnecessary call.